frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -46,69 +66,115 @@ const getClusterUpdateStatus = (cv: K8sResourceKind): ClusterUpdateStatus => {
 
 const getIconClass = (status: ClusterUpdateStatus) => {
   return {
-    [ClusterUpdateStatus.UpToDate]: 'pficon pficon-ok text-success',
-    [ClusterUpdateStatus.UpdatesAvailable]: 'fa fa-arrow-circle-up text-success',
-    [ClusterUpdateStatus.Updating]: 'pficon pficon-in-progress',
-    [ClusterUpdateStatus.Failing]: 'pficon pficon-error-circle-o text-danger',
-    [ClusterUpdateStatus.ErrorRetrieving]: 'pficon pficon-error-circle-o text-danger',
+    [ClusterUpdateStatus.UpToDate]: 'pficon pficon-ok',
+    [ClusterUpdateStatus.UpdatesAvailable]: 'fa fa-arrow-circle-o-up',
+    [ClusterUpdateStatus.Updating]: 'fa-spin pficon pficon-in-progress',
+    [ClusterUpdateStatus.Failing]: 'pficon pficon-error-circle-o',
+    [ClusterUpdateStatus.ErrorRetrieving]: 'pficon pficon-error-circle-o',
   }[status];
 };
 
-const UpdateStatus: React.SFC<UpdateStatusProps> = ({cv}) => {
+const FailedConditionAlert = ({message, condition}) => <div className="alert alert-danger">
+  <i className="pficon pficon-error-circle-o" aria-hidden="true" /> <strong>{message}</strong> {condition.message}
+</div>;
+
+const UpdateInProgressAlert = () => <div className="alert alert-info">
+  <i className="pficon pficon-info" aria-hidden={true} />
+  Cluster update in progress.
+  &nbsp;
+  <Link to="/settings/cluster/clusteroperators">
+    View detailed progress.
+  </Link>
+</div>;
+
+const UpdatesAvailableAlert = ({cv}) => <div className="alert alert-info">
+  <i className="pficon pficon-info" aria-hidden={true} />
+  Cluster update is available.
+  <Button bsStyle="link" onClick={()=> (launchUpdateModal(cv))}>
+    Update Now
+  </Button>
+</div>;
+
+const ClusterVersionAlerts: React.SFC<ClusterVersionAlertsProps> = ({cv, updatesAvailable}) => {
+  const conditions = _.get(cv, 'status.conditions', []);
+  const retrievedUpdatesFailedCondition = !MOCK_CLUSTER_UPDATE && _.find(conditions, { type: 'RetrievedUpdates', status: 'False' });
+  const isFailingCondition = !MOCK_CLUSTER_UPDATE && _.find(conditions, { type: 'Failing', status: 'True' });
   const status = getClusterUpdateStatus(cv);
+
+  return <React.Fragment>
+    { status === ClusterUpdateStatus.Updating && <UpdateInProgressAlert /> }
+    { updatesAvailable && <UpdatesAvailableAlert cv={cv} /> }
+    { isFailingCondition && <FailedConditionAlert message="Upgrade is failing." condition={isFailingCondition} />}
+    { retrievedUpdatesFailedCondition && <FailedConditionAlert message="Could not retrieve updates." condition={retrievedUpdatesFailedCondition} />}
```

Comment:
I think this is a runtime error if every one of these conditions evaluates to false.

Reply:
I think it will be okay since they are all wrapped in a React.Fragment. I'll test to make sure though.

Reply:
I'm pretty sure it's still an error :/